browser/src/Editor/SimpleNeovimEditor.tsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * NeovimEditor.ts
+ *
+ * IEditor implementation for Neovim
+ */
+
+import * as React from "react"
+
+import { IncrementalDeltaRegionTracker } from "./../DeltaRegionTracker"
+import { NeovimInstance } from "./../neovim"
+import { CanvasRenderer, DOMRenderer, INeovimRenderer } from "./../Renderer"
+import { NeovimScreen } from "./../Screen"
+
+import * as Config from "./../Config"
+
+// import { PluginManager } from "./../Plugins/PluginManager"
+
+import { Keyboard } from "./../Input/Keyboard"
+import { IEditor } from "./Editor"
+
+import { NeovimRenderer } from "./NeovimRenderer"
+
+export class DummyPluginManager {
+    public getAllRuntimePaths(): string[] {
+        return []
+    }
+
+    public startPlugins(neovimInstance: NeovimInstance): void {
+        // no-op
+    }
+}
+
+export class SimpleNeovimEditor implements IEditor {
+
+    private _neovimInstance: NeovimInstance
+    private _deltaRegionManager: IncrementalDeltaRegionTracker
+    private _renderer: INeovimRenderer
+    private _screen: NeovimScreen
+
+    private _pendingTimeout: any = null
+    private _pendingAnimationFrame: boolean = false
+
+    constructor(
+        private _config: Config.Config = Config.instance(),
+    ) {
+        // TODO: How to get rid of this?
+        this._neovimInstance = new NeovimInstance(new DummyPluginManager(), 100, 100)
+        this._neovimInstance.setInitVim("C:/oni/test.vim")
+        this._deltaRegionManager = new IncrementalDeltaRegionTracker()
+        this._screen = new NeovimScreen(this._deltaRegionManager)
+
+        this._renderer = this._config.getValue("editor.renderer") === "canvas" ? new CanvasRenderer() : new DOMRenderer()
+
+        this._render()
+
+        this._config.registerListener(() => this._onConfigChanged())
+
+        this._neovimInstance.on("action", (action: any) => {
+            this._renderer.onAction(action)
+            this._screen.dispatch(action)
+
+            this._scheduleRender()
+
+            if (!this._pendingTimeout) {
+                this._pendingTimeout = setTimeout(() => this._onUpdate(), 0)
+            }
+        })
+
+        const keyboard = new Keyboard()
```

Comment:
I'm assuming when you say extract keyboard you mean have detached from an editor and whichever editor has focus will receive the events right?

Sounds like a nice polymorphic opportunity for editors.

The difference would be instead of an interface have an abstract, and then have a static instance/private instance for _currentEditor, and that way you don't have to change who keyboard is sending to because it can just be _currentEditor.sendKeys() and then _currentEditor can process it based on it's overrided method but can also pull common code from AbstractEditor, thoughts?